crafting_combinator/script/rc.lua

```diff
@@ -140,13 +155,37 @@ function _M:find_ingredients_and_products(forced)
 end
 
 
+function _M:find_machines(forced)
+	local recipe, input_count = recipe_selector.get_recipe(self.entity, nil, defines.circuit_connector_id.combinator_input)
+
+	if self.recipe ~= recipe or forced then
+		self.recipe = recipe
+		local params = {}
+		if recipe and recipe.category then
+			for i, mac in pairs(global.rc.machines[recipe.category] or {}) do
+				local mac_res = self.entity.force.recipes[mac]
```

Comment:
This wouldn't work for machines that have a different name than the recipe that produces them. This should probably check if there is some recipe that produces the item that places the machine instead.